### PR TITLE
Workaround for negative log in simple heat exchanger kA function

### PR DESCRIPTION
**Pull request by jbueck**
The simple heat exchanger kA function runs into a negative log error, when exactly one of the two temperature differences is negative. This is a workaround that uses a mean difference in those cases. In the calc_parameters function in simple_heat_exchanger an if clause shall prevent a manually set kA parameter to be overwritten.

### CODE REVIEW

File: src/tespy/components/heat_exchangers.py [hunk 1]
```diff
@@ -640,7 +640,11 @@ def kA_func(self):
         ttd_1 = T_mix_ph(i, T0=self.inl[0].T.val_SI) - self.Tamb.val_SI
         ttd_2 = T_mix_ph(o, T0=self.outl[0].T.val_SI) - self.Tamb.val_SI
 
-        if ttd_1 > ttd_2:
+        # If one temperature difference is negative: use mean difference to
+        # avoid negative logarithm otherwise use lmtd
+        if (ttd_1 / ttd_2) < 0:
+            td_log = (ttd_2 + ttd_1) / 2
```
[the remark this answers is not in the document]
Comment (jbueck) [reply src/tespy/components/heat_exchangers.py:646]: No, it does not and this is not what it shall do, as this is a plausible thing to happen. It avoids a negative value in the logarithm function, that would throw en error.

File: src/tespy/components/heat_exchangers.py [hunk 2]
```diff
@@ -847,17 +851,22 @@ def calc_parameters(self):
...
+                    td_log = (ttd_2 + ttd_1) / 2
+                elif ttd_1 > ttd_2:
+                    td_log = (ttd_1 - ttd_2) / np.log(ttd_1 / ttd_2)
+                elif ttd_1 < ttd_2:
+                    td_log = (ttd_2 - ttd_1) / np.log(ttd_2 / ttd_1)
+                else:
+                    td_log = 0
 
-            self.kA.val = abs(i[0] * (o[2] - i[2]) / td_log)
+                self.kA.val = abs(i[0] * (o[2] - i[2]) / td_log)
```
[the remark this answers is not in the document]
Comment (jbueck) [reply src/tespy/components/heat_exchangers.py:869]: I just thought that this is unnecessary as this value is set fix and does not need to be recalculated. I guess your proposal to compare the calculated value to the given one is even better.

File: src/tespy/components/heat_exchangers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fwitte) [reply src/tespy/components/heat_exchangers.py:646]: Well misread the code, thanks :+1:

File: src/tespy/components/heat_exchangers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fwitte) [reply src/tespy/components/heat_exchangers.py:869]: I will need to change it in all component classes, after that we can merge.

**Pull request merged**